ipfx/nwb_reader.py

```diff
@@ -77,9 +77,11 @@ def read_sweep_from_source(source):
             elif "sweep_number" in timeseries.attrs:
                 real_sweep_number = timeseries.attrs["sweep_number"]
 
-        if assumed_sweep_number is not None and assumed_sweep_number != real_sweep_number:
-            warnings.warn("Sweep number mismatch (real: {} vs assumed: {})".format
-                          (real_sweep_number, assumed_sweep_number))
+        if assumed_sweep_number is not None and real_sweep_number is not None:
+
+            if assumed_sweep_number != real_sweep_number:
+                warnings.warn("Sweep number mismatch (real: {} vs assumed: {})".format
+                              (real_sweep_number, assumed_sweep_number))
```

Comment:
Change looks good although it takes a second to appreciate it.

Reply:
Thanks for a careful review. I hope I addressed all of your points.